5_evaluate/src/eval_utility_fxns.R

```diff
@@ -172,24 +184,25 @@ calc_rmse <- function(eval_pred_obs, grouping_var, depth_class) {
 #' @param plot_df a tibble of the matched model predictions and observations,
 #' along with grouping variables for evaluation
 #' @param num_eval_sites The number of unique evaluation sites 
-#' @param driver the name of the driver used to generate the
-#' model predictions
+#' @param driver_type the type of driver used to generate the model
+#' predictions (NLDAS or GCM)
 #' @param y_var the variable for the y-axis of the plot
 #' @param y_label the label for the y-axis of the plot
 #' @param x_var the variable for the x-axis of the plot
-#' @faceting_variable variable to use for faceting the plot
+#' @facet_column_variable variable to use for faceting the columns of the plot
+#' @facet_row_variable variable to use for faceting the rows of the plot
 #' @param outfile The filepath for the exported png
 #' @return The filepath of the exported png 
-plot_evaluation_barplot <- function(plot_df, num_eval_sites, driver, y_var, y_label, x_var, faceting_variable, 
-                                    outfile, plot_dpi = 300, plot_width = 10, plot_height = 8) {
+plot_evaluation_barplot <- function(plot_df, num_eval_sites, driver_type, y_var, y_label, x_var, facet_column_variable, 
+                                    facet_row_variable, outfile, plot_dpi = 300, plot_width = 10, plot_height = 8) {
   bar_plot <- plot_df %>%
     ggplot(aes(x = get(x_var), y = get(y_var))) +
     geom_col(fill='cadetblue3', color='cadetblue4') +
-    labs(title = paste(sprintf("%s predictions: %s by %s", driver, y_var, x_var),
+    labs(title = paste(sprintf("%s predictions: %s by %s", driver_type, y_var, x_var),
                        sprintf("Total # of evaluation sites: %s", num_eval_sites), sep ='\n'), 
          x=sprintf("%s", x_var), 
          y=sprintf("%s (\u00b0C)", y_label)) +
-    facet_wrap(~get(faceting_variable), nrow = 2) +
+    facet_grid(get(facet_row_variable) ~ get(facet_column_variable)) +
```

Comment:
Woah! I've never used `get()`. Do you know if it is equivalent to `!!facet_row_variable` ?

Reply:
I'm not totally certain how it differs, but I think it does, b/c I just tested `!!facet_row_variable` locally and I get an 'invalid argument type' error. 🤷‍♀️. I stumbled upon `get()` when working on plotting functions for the PGDL work, so it's what I thought to use here.